### DISCUSSION TITLE
Project Roadmap

### DISCUSSION THREAD

**Discussion by saddam213** [saddam213/OnnxStack#27] [General]
Rough roadmap, will update

- [ ] Upscaler
- [ ] Safety Filter
- [ ] LoRA
- [x] SDXL Pipeline
- [x] InstaFlow Pipeline
- [ ] Controlnet Pipeline
- [ ] AnimatedDiff
- [ ] Image Recognition
- [ ] Image Classification

**Comment by saddam213** [comment]
1. Is almost done, but output result is cloudy

2. Not Started

3. Started, lost, research links below
https://medium.com/openvino-toolkit/enable-lora-weights-with-stable-diffusion-controlnet-pipeline-94fb06c86ea2
https://github.com/microsoft/onnxruntime/blob/main/csharp/test/Microsoft.ML.OnnxRuntime.Tests.Common/TrainingTest.cs

4. Complicated but confident, but waiting on GPU with more VRAM

5. New Pipeline, Diffusers and perhaps schedulers

6, 7 Should be easy to implement now as both `OnnxStack.Core` and `OnnxStack.StableDiffuion` have matured

**Comment by Amin456789** [comment]
nice roadmap! thanks for ur hard work!

**Comment by Amin456789** [comment]
1.maybe u should try other models, there are plenty of them, see which one will work

2.please make this one turn off by default so anyone wanna have it can enable it in settings

3.can't comment, but u can do it! work on lcm first it will be easier to impelement maybe? the repo fastsdcpu they now support lora-lcm maybe u can get help from their openvino code

4.nice! cant wait for this one, please make sure to bring other options such as context size, i tested this and lower context needed lower time to generate [at least for my cpu back then]
also, it will be great if u find a way to connect lcm to animatediff for much faster generation [people got some solutions in comfyui with workflows i guess]

5.i think i will stick with 1.5 and lcm for now but it will be great for whoever is into SDXL!

6 and 7. can't comment, don't know what they are, but if they are something like image to text = CLIP, then great!

**Reply by saddam213** [reply]
Just so you know a bit more about 6 & 7 here is a few screen shots that explain it simply

### Image Recognition
![image](https://github.com/saddam213/OnnxStack/assets/4353684/27392e16-11fc-4512-9327-a256786e3dc5)

### Image Classification 
![out](https://github.com/saddam213/OnnxStack/assets/4353684/4be8c8bc-204a-488f-8495-bbef2944cad2)

**Reply by Amin456789** [reply]
thats great!

**Comment by AshD** [comment]
This is a great roadmap :-)

One addition to consider is SadTalker
https://huggingface.co/vinthony/SadTalker-V002rc
https://github.com/OpenTalker/SadTalker

I am hoping they create an Onnx version of their model.

Regards,
Ash

**Comment by saddam213** [comment]
SDXL in the house!!

![DreamShaper_XL_624461087_Euler](https://github.com/saddam213/OnnxStack/assets/4353684/2d5e5d89-772d-4478-8b76-4aff645ea694)

**Reply by AshD** [reply]
Cool
